tests/planner.spec.ts

```diff
@@ -0,0 +1,142 @@
+import { test, expect, Page } from "@playwright/test";
+
+const testForArcanistPlanning = (
+  ...arcanists: {
+    name: string
+    insight: [number, number]
+    level: [number, number]
+    resonance: [number, number]
+    expectMaterials: {
+      name: string
+      count: string
+    }[]
+  }[]
+) => {
+  return async ({ page }: { page: Page }) => {
+    const setupStatus = async (blockName: string, status: [string, string]) => {
+      if (status.length !== 2) { return }
+
+      const settingBlock = page.locator('.edit-overlay')
+        .locator(`:below(:text("${blockName}"))`)
+        .first()
+
+      for (let i = 0; i < status.length; i++) {
+        await settingBlock.getByRole('button').nth(i).click()
+        const fromBlock = settingBlock.locator('div:below(button)')
+          .filter({ has: page.getByRole('button') })
+          .first()
+        await fromBlock.getByRole('button', { name: status[i], exact: true }).click()
+      }
+    }
+
+    for (let i = 0; i < arcanists.length; i++) {
+      const arcanist = arcanists[i]
+      await page.getByRole("button", { name: "Add Arcanist" }).click()
+      await page.getByLabel('Show Unreleased Arcanists').check()
+      await page.getByText(arcanist.name).click()
+
+      await setupStatus('CURRENT LEVEL', [
+        arcanist.insight[0] ? `Option ${arcanist.insight[0]}` : '', `${arcanist.level[0]}`
+      ])
+      await setupStatus('GOAL LEVEL', [
+        arcanist.insight[1] ? `Option ${arcanist.insight[1]}` : '', `${arcanist.level[1]}`
+      ])
+      await setupStatus('RESONANCE', arcanist.resonance.map(item => `${item}`) as [string, string])
+
+
+      const materialsBlock = page.locator('div:below(button:text("Save"))')
+        .filter({ has: page.getByRole('img') })
+        .first()
+      await Promise.all(
+        arcanist.expectMaterials.map(material => expect(
+          materialsBlock.locator(`[data-tip="${material.name}"]`).getByText(material.count)
+        ).toBeVisible())
+      )
+
+      await page.getByRole('button', { name: 'Save' }).click()
+    }
+
+    await expect(page).toHaveScreenshot();
```

Comment:
The use of **screenshots** for testing is generally recommended in **stable UI scenarios**. Within the kblm project, I do not recommend using screenshots for testing due to the frequent changes in the UI, which often lead to displacements. When faced with such situations, we have the following options:

1. Frequently update the screenshot.
2. Set certain ratios to ensure the test passes.

The first option is not very practical for open-source projects... It demands that other developers possess the same knowledge to update the screenshot before making any related changes.
The second option might lead to the screenshot test losing its purpose.

Ask yourselves: *What do we really need to test?*
If it is necessary, then let's add specific expect statements to test for that specific UI 👌

Reply:
My purpose was to test the planning result, like "farm dusts from stage: The Poussiere VI".
It's hard to calculate which stage to farm. But as you said, maybe screenshots are not a good solution to our project for now.
Shall we write another test only for the stage block, write a hard-code expection for only 1 planning result?

Reply:
Or if you have other good idea, please let me know.

Reply:
If you just want to check "farm dusts from stage: The Poussiere VI," I think it would be better to just set some basic planner settings and check whether the Resources block is visible, and some ui detail in there.